### DISCUSSION TITLE
Iol / Ioll2 Vlans & vrnetlab

### DISCUSSION THREAD

**Discussion by DanPartelly** [ipspace/netlab#1470] [Show and tell]
Roman's fork of vrnetlab switched today to my fork of iouyap with commit cea756c . 

The practical importance of this for us is that any iol* image built with master should have functional vlans,trunking, spanning tree and should pass integration tests. (Both L2 and L3 images need a rebuild)

**Comment by ipspace** [comment]
Great to hear about the vrnetlab fix. Will try to rebuild my containers.

However, I don't think bridging works correctly on IOL (I couldn't configure the IEEE STP).

**Comment by ipspace** [comment]
I rebuilt the containers and tried the simplest multi-node test (`vlan/31-vlan-bridge-trunk.yml`). Works like a charm on IOLL2 (great job, thanks a million!!!) but not on IOL (as expected).

I'm running the IOLL2 VLAN integration tests, once they pass we can fix the device settings/documentation/caveats.

**Reply by DanPartelly** [reply]
> I rebuilt the containers and tried the simplest multi-node test (`vlan/31-vlan-bridge-trunk.yml`). Works like a charm on IOLL2 (great job, thanks a million!!!) but not on IOL (as expected).
> 

Thanks! Yeah, it doest sends any BPDUs .  What works on iol are  scenarios like 51-vlan-routed-trunk.  This begs the question, keep it vlan enabled or not ?

**Reply by jbemmel** [reply]
SRLinux doesn’t support STP either

**Reply by ipspace** [reply]
> Thanks! Yeah, it doest sends any BPDUs.

That should not be a show-stopper. Either it puts the links into blocked state because of missing STP code, or it doesn't forward packets at all.

> What works on iol are scenarios like 51-vlan-routed-trunk. This begs the question, keep it vlan enabled or not ?

I will try to figure out whether there's a simple way to limit the functionality to routed subinterfaces. If I can't get that done, I'll disable the VLAN feature for IOL.

**Comment by ipspace** [comment]
Managed to disable VLAN/SVI interfaces in IOL device settings. The only scenarios that can be implemented are routed subinterfaces, and they all work ;) https://tests.netlab.tools/_html/iol-clab-vlan

Thanks again @DanPartelly for a phenomenal integration job! 🎉🎉🎉

**Reply by DanPartelly** [reply]
Glad we got to keep those enabled!!

> Managed to disable VLAN/SVI interfaces in IOL device settings. The only scenarios that can be implemented are routed subinterfaces, and they all work ;)
